### DISCUSSION TITLE
feat: add index.js files for sub-folders (f.e. solid-js/web)

### DISCUSSION THREAD

**Discussion by trusktr** [solidjs/solid#799] [Ideas]
Some ESM systems (f.e. the unpkg.com CDN) don't yet understand package.json `exports` fields, but they do look for index.js files.

https://github.com/mjackson/unpkg/issues/286

To make solid-js code compatible in more systems, we could add one-line `index.js` files to sub-folders that don't have index files.

As an example, `import('//unpkg.com/solid-js@1.2.5/web?module')` won't work because it looks for `unpkg.com/solid-js@1.2.5/web/index.js`, and totally ignore `exports` rules in package.json.

As we already have exports rules and systems like Node.js and Webpack 5+ will honor those rules, adding an `index.js` file will not break those systems because they will continue to map themselves to files in `dist/`.

I believe adding `index.js` files will be backward compatible, and allow solid-js to be compatible in more places.

The files will look like this:

```js
// solid-js/web/index.js
export * from './dist/web.js'
```

```js
// solid-js/html/index.js
export {default as default} from './dist/html.js'
```

Ready to test it out and make a PR.

**Comment by ryansolid** [comment]
They ignore nested package.json main/module fields as well? I mean I guess it can't hurt. Just unfortunate the 2 common ways this is done(the recommended, and the legacy) don't work and we would need to add a 3rd way.
